sdk/storage/azdatalake/directory/client_test.go

```diff
@@ -767,6 +769,72 @@ func (s *RecordedTestSuite) TestDeleteDirWithNilAccessConditions() {
 	_require.NotNil(resp)
 }
 
+func (s *UnrecordedTestSuite) TestDeleteDirWithPaginatedDelete() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	user := "user"
+	readWriteExecutePermission := "rwx"
+
+	objectId := "48965e63-fd19-4438-b9d0-dc585a8baf57"
+	accountName, accountKey := testcommon.GetGenericAccountInfo(testcommon.TestAccountDatalake)
+
+	filesystemName := testcommon.GenerateFileSystemName(testName)
+	fsClient, err := testcommon.GetFileSystemClient(filesystemName, s.T(), testcommon.TestAccountDatalake, nil)
+	_require.NoError(err)
+	defer testcommon.DeleteFileSystem(context.Background(), _require, fsClient)
+
+	_, err = fsClient.Create(context.Background(), nil)
+	_require.NoError(err)
+
+	rootDirectory := fsClient.NewDirectoryClient("/")
+
+	dirName := testcommon.GenerateDirName(testName)
+	dirURL := "https://" + accountName + ".dfs.core.windows.net/" + filesystemName + "/" + dirName
+	credential, err := azdatalake.NewSharedKeyCredential(accountName, accountKey)
+	_require.NoError(err)
+
+	dirClient, err := directory.NewClientWithSharedKeyCredential(dirURL, credential, nil)
+	_require.NoError(err)
+
+	resp, err := dirClient.Create(context.Background(), nil)
+	_require.NoError(err)
+	_require.NotNil(resp)
+
+	for i := 0; i < 5; i++ {
```

Comment:
Are we getting continuation token for only 5 files here?

Reply:
My bad, had reduced to 5 for testing. Reverted it back to > 5k files